src/main/host/host.rs

```diff
@@ -569,6 +576,10 @@ impl Host {
 
 impl Drop for Host {
     fn drop(&mut self) {
+        let default_addr = self.default_address.borrow(&self.root).ptr();
+        if !default_addr.is_null() {
+            unsafe { cshadow::address_unref(default_addr) };
+        }
         unsafe { cshadow::hostc_unref(self.chost()) };
     }
```

Comment:
nit: maybe also set `default_address` to `NULL`, just in case for example `hostc_unref` were to call something that calls `host_getDefaultAddress` which then dereferences the dangling pointer.

Reply:
nvm, i forgot the C host doesn't have access to the rust host